plugin/v2/plugin.go

```diff
@@ -29,10 +29,12 @@ type Plugin struct {
 const defaultPluginRuntimeDestination = "/run/docker/plugins"
 
 // ErrInadequateCapability indicates that the plugin did not have the requested capability.
-type ErrInadequateCapability string
+type ErrInadequateCapability struct {
+	cap string
+}
 
-func (cap ErrInadequateCapability) Error() string {
-	return fmt.Sprintf("plugin does not provide %q capability", cap)
```

Comment:
I'm following the convention used in https://github.com/docker/docker/blob/master/client/errors.go. This is absolutely fine as well.